File: src/main/java/com/es2/vadebicicleta/externo/commons/repository/manager/LongIdManager.java
package com.es2.vadebicicleta.externo.commons.repository.manager;


import org.springframework.stereotype.Component;

import java.util.HashMap;
import java.util.Map;

@Component
public class LongIdManager {
    private static Map<Class<?>, Long> idCounters;

    public LongIdManager(){
        idCounters = new HashMap<>();
    }

    public Long generateId(Class<?> repository, LongGenerationStrategy strategy){
        if(LongGenerationStrategy.IDENTITY.equals(strategy)){
            return identity(repository);
        }
        else{
            return null;
        }
    }

    private Long identity(Class<?> repository){
       Long idCounter = idCounters.get(repository);
       if(idCounter == null){
           idCounter = 0L;

       }else {
           idCounter++;
       }
       idCounters.put(repository, idCounter);

       return idCounter;
    }

    public enum LongGenerationStrategy{
        IDENTITY
    }

}

File: src/main/kotlin/com/es2/vadebicicleta/externo/commons/repository/manager/LongIdManager.kt
package com.es2.vadebicicleta.externo.commons.repository.manager

import org.springframework.stereotype.Component

@Component
class LongIdManager {
    private val idCounters: MutableMap<Class<*>, Long> = mutableMapOf()

    fun generateId(repository: Class<*>, strategy: LongGenerationStrategy): Long {
        return when(strategy){
            LongGenerationStrategy.IDENTITY -> identity(repository)
        }
    }

    private fun identity(repository: Class<*>): Long {
        var idCounter = idCounters[repository]
        if (idCounter == null) {
            idCounter = 0L
        } else {
            idCounter++
        }
        idCounters[repository] = idCounter
        return idCounter
    }

    enum class LongGenerationStrategy {
        IDENTITY
    }
}
